build-system/tasks/helpers.js

```diff
@@ -0,0 +1,746 @@
+    const copy = Object.create(options);
+    copy.watch = false;
+    $$.watch(path + '/*', function() {
+      buildExperiments(copy);
+    });
+  }
+
+  // Build HTML.
+  const html = fs.readFileSync(htmlPath, 'utf8');
+  const minHtml = html.replace('/dist.tools/experiments/experiments.js',
+      `https://${hostname}/v0/experiments.js`);
+  gulp.src(htmlPath)
+      .pipe($$.file('experiments.cdn.html', minHtml))
+      .pipe(gulp.dest('dist.tools/experiments/'));
+
+  // Build JS.
+  const js = fs.readFileSync(jsPath, 'utf8');
+  const builtName = 'experiments.max.js';
+  const minifiedName = 'experiments.js';
+  return toPromise(gulp.src(path + '/*.js')
+      .pipe($$.file(builtName, js))
+      .pipe(gulp.dest('build/experiments/')))
+      .then(function() {
+        return compileJs(
+            './build/experiments/', builtName, './dist.tools/experiments/', {
+              watch: false,
+              minify: options.minify || argv.minify,
+              includePolyfills: true,
+              minifiedName,
+              preventRemoveAndMakeDir: options.preventRemoveAndMakeDir,
+              checkTypes: options.checkTypes,
+            });
+      });
+}
+
+/**
+ * Build ALP JS.
+ *
+ * @param {!Object} options
+ */
+function buildAlp(options) {
+  options = options || {};
+  return compileJs('./ads/alp/', 'install-alp.js', './dist/', {
+    toName: 'alp.max.js',
+    watch: options.watch,
+    minify: options.minify || argv.minify,
+    includePolyfills: true,
+    minifiedName: 'alp.js',
+    preventRemoveAndMakeDir: options.preventRemoveAndMakeDir,
+  });
+}
+
+/**
+ * Build Examiner JS.
+ *
+ * @param {!Object} options
+ */
+function buildExaminer(options) {
+  options = options || {};
+  return compileJs('./src/examiner/', 'examiner.js', './dist/', {
+    toName: 'examiner.max.js',
+    watch: options.watch,
+    minify: options.minify || argv.minify,
+    includePolyfills: true,
+    minifiedName: 'examiner.js',
+    preventRemoveAndMakeDir: options.preventRemoveAndMakeDir,
+  });
+}
+
+/**
+ * Build web worker JS.
+ *
+ * @param {!Object} options
+ */
+function buildWebWorker(options) {
+  const opts = Object.assign({}, options);
+  return compileJs('./src/web-worker/', 'web-worker.js', './dist/', {
+    toName: 'ww.max.js',
+    minifiedName: 'ww.js',
+    includePolyfills: true,
+    watch: opts.watch,
+    minify: opts.minify || argv.minify,
+    preventRemoveAndMakeDir: opts.preventRemoveAndMakeDir,
+  });
+}
+
+/**
+ *Creates directory in sync manner
+ *
+ * @param {string} path
+ */
+function mkdirSync(path) {
+  try {
+    fs.mkdirSync(path);
+  } catch (e) {
+    if (e.code != 'EEXIST') {
+      throw e;
+    }
+  }
+}
+
+/**
+ * Returns a promise for readable
+ *
+ * @param {*} readable
+ * @return {Promise}
+ */
+function toPromise(readable) {
+  return new Promise(function(resolve, reject) {
+    readable.on('error', reject).on('end', resolve);
+  });
+}
+
+exports.buildAlp = buildAlp;
```

Comment:
For multiline exports like this, don't we usually do:
```javascript
exports = {
  buildAlp,
  buildExaminer,
  ...
}
```

Reply:
That didn't work. Will punt for now, and revisit when I further refactor the code in `helpers.js`.
```js
[14:06:32] TypeError: printConfigHelp is not a function
    at Gulp.dist (/usr/local/google/home/rsimha/src/amphtml/build-system/tasks/dist.js:74:5)
    at module.exports (/usr/local/google/home/rsimha/src/amphtml/node_modules/orchestrator/lib/runTask.js:34:7)
    at Gulp.Orchestrator._runTask (/usr/local/google/home/rsimha/src/amphtml/node_modules/orchestrator/index.js:273:3)
    at Gulp.Orchestrator._runStep (/usr/local/google/home/rsimha/src/amphtml/node_modules/orchestrator/index.js:214:10)
    at /usr/local/google/home/rsimha/src/amphtml/node_modules/orchestrator/index.js:279:18
    at finish (/usr/local/google/home/rsimha/src/amphtml/node_modules/orchestrator/lib/runTask.js:21:8)
    at module.exports (/usr/local/google/home/rsimha/src/amphtml/node_modules/orchestrator/lib/runTask.js:60:3)
    at Gulp.Orchestrator._runTask (/usr/local/google/home/rsimha/src/amphtml/node_modules/orchestrator/index.js:273:3)
    at Gulp.Orchestrator._runStep (/usr/local/google/home/rsimha/src/amphtml/node_modules/orchestrator/index.js:214:10)
    at Gulp.Orchestrator.start (/usr/local/google/home/rsimha/src/amphtml/node_modules/orchestrator/index.js:134:8)
    at /usr/local/google/home/rsimha/.config/yarn/global/node_modules/gulp-cli/lib/versioned/^3.7.0/index.js:72:20
    at process._tickCallback (internal/process/next_tick.js:61:11)
```

Reply:
Aha. It's supposed to be `module.exports = {...};`, not `exports = {...};`. Done in #22125.